### PR TITLE
INTMDB-611: Added Profile to teams resource

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

This PR updates teams to get api keys from AWS secret manager.

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments
Tested by running `cfn test` locally

```
8 passed, 4 skipped, 12 deselected, 1 warning in 53.23s
```

### CODE REVIEW

File: cfn-resources/teams/test/inputs_1_create.template.json [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
-    "Usernames": ["r.dharmendran@gmail.com"]
+  "Profile": "default",
+  "RoleNames": [
+    "GROUP_CLUSTER_MANAGER"
+  ],
+  "OrgId": "",
+  "TeamId": "",
+  "Name": "testTeam",
+  "Usernames": [
+    "r.dharmendran@gmail.com"
```
Comment (govardhanpagidi) [inline cfn-resources/teams/test/inputs_1_create.template.json:10]: can we remove this email id?

File: examples/teams/teams.json [hunk 2]
```diff
@@ -37,17 +43,18 @@
...
     }
   },
   "Outputs": {
-    "Teams" : {
+    "Teams": {
       "Description": " teams",
-      "Value" : { "Ref" : "Teams" }
+      "Value": {
+        "Ref": "Teams"
+      }
```
Comment (govardhanpagidi) [inline examples/teams/teams.json:57]: if you have tested,  will { "Ref":"Teams"} return all the properties under PrimaryIdentifier?

**Review by govardhanpagidi — APPROVED** [review]
LGTM, Minor comments but not show stoppers.

File: cfn-resources/teams/test/inputs_1_create.template.json (hunk 1 above)
Comment (andreaangiolillo) [reply cfn-resources/teams/test/inputs_1_create.template.json]: updated

File: examples/teams/teams.json (hunk 2 above)
Comment (andreaangiolillo) [reply examples/teams/teams.json:57]: testing now

Comment (andreaangiolillo) [reply examples/teams/teams.json:57]: I only have "/properties/TeamId" in the private identifier so it returned only the teamId

Screenshot 2023-02-22 at 09 49 33

**Comment by andreaangiolillo** [comment]
Tested the resource by releasing in the private registry and verified that the team was created
Screenshot 2023-02-22 at 09 51 05
Screenshot 2023-02-22 at 09 49 33

File: examples/teams/teams.json (hunk 2 above)
Comment (andreaangiolillo) [reply examples/teams/teams.json:57]: yes, I see the TeamId and the profile name

Screenshot 2023-02-22 at 10 03 39

**Comment by andreaangiolillo** [comment]
I tested the new resource by releasing it in the private registry and checked that the team was generated.

File: cfn-resources/teams/cmd/resource/resource.go [hunk 3]
```diff
@@ -286,28 +293,26 @@ func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.P
...
 
-	// Create mongo DB client
-	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
-	if err != nil {
-		_, _ = logger.Warnf(constants.ErrorCreateMongoClient, err)
-		return progressevents.GetFailedEventByCode(fmt.Sprintf("Failed to Create Client : %s", err.Error()),
-			cloudformation.HandlerErrorCodeInvalidRequest), err
+	client, handlerError := util.NewMongoDBClient(req, currentModel.Profile)
+	if handlerError != nil {
+		_, _ = logger.Warnf("CreateMongoDBClient error: %v", handlerError)
```
Comment (maastha) [inline cfn-resources/teams/cmd/resource/resource.go:302]: NewMongoDBClient method already logs errors, is this still needed? if yes, can we move it to NewMongoDBClient instead?

**Review by maastha — APPROVED** [review]
Comment is non-blocking, otherwise LGTM

File: cfn-resources/teams/cmd/resource/resource.go (hunk 3 above)
Comment (colm-quinn) [reply cfn-resources/teams/cmd/resource/resource.go:302]: If you're updating, can you also consider moving the profile update into the helper method. We have a lot of repeated code here that can be handled by the client.

**Pull request merged**